zkevm-circuits/src/evm_circuit/execution/end_tx.rs

```diff
@@ -254,10 +253,8 @@ impl<F: Field> ExecutionGadget<F> for EndTxGadget<F> {
         let current_cumulative_gas_used: u64 = if tx.id == 1 {
             0
         } else {
-            let rw = &block.rws[(
-                RwTableTag::TxReceipt,
-                (tx.id - 1) * TxReceiptFieldTag::COUNT - 1,
-            )];
+            let idx = if tx.id == 2 { 2 } else { (tx.id - 2) * 4 + 2 };
```

Comment:
I guess you are testing 3 or more tx scenarios , thanks for catching ! better to add unit tests to cover it as current only have two tx cases in end tx's unit test.   
suggest to following change to avoid hard code `4` multiplier  :  
`let idx = if tx.id == 2 { 2 } else { (tx.id - 2) * (TxReceiptFieldTag::COUNT + 1 ) + 2 };`

Reply:
seems it can be written as `let idx = (tx.id - 2) * (TxReceiptFieldTag::COUNT + 1) + 2;`

Reply:
good to me , I like it !!

Reply:
`(tx.id - 1) * TxReceiptFieldTag::COUNT - 1` 

=> 

`(tx.id - 1) * (TxReceiptFieldTag::COUNT + 1) - 2`